ocs_ci/ocs/node.py

```diff
@@ -82,52 +83,45 @@ def wait_for_nodes_status(node_names=None, status=constants.NODE_READY, timeout=
             for node in sample:
                 if not node_names:
                     return True
-                for status_condition in node.get('status').get('conditions'):
-                    if status == constants.NODE_NOT_READY:
-                        message = 'Kubelet stopped posting node status'
-                        if message in status_condition.get('message'):
-                            node_names.remove(node.get('metadata').get('name'))
-                            break
-                    else:
-                        if 'True' in status_condition.get('status'):
-                            log.info(
-                                f"The following nodes are still not "
-                                f"in {status} status: {node_names}"
-                            )
-                            if status_condition.get('type') == status:
-                                node_names.remove(node.get('metadata').get('name'))
-                                break
+                if node.ocp.get_resource_status(node.name) == status:
+                    node_names.remove(node.name)
+                    break
     except TimeoutExpiredError:
         log.error(f"The following nodes haven't reached status {status}: {node_names}")
         return False
 
 
-def unschedule_nodes(nodes):
+def unschedule_nodes(node_names):
     """
     Change nodes to be unscheduled
 
     Args:
-        nodes (list): The OCP objects of the nodes
+        node_names (list): The names of the nodes
 
     """
-    for node in nodes:
-        node.exec_oc_cmd(f"adm cordon {node.get('metadata').get('name')}")
-    assert wait_for_nodes_status(nodes, status=constants.NODE_SCHEDULING_DISABLED), (
-        f"Not all nodes reached status {constants.NODE_SCHEDULING_DISABLED}"
-    )
+    ocp = OCP(kind='node')
+    for node_name in node_names:
+        ocp.exec_oc_cmd(f"adm cordon {node_name}")
+    assert wait_for_nodes_status(
```

Comment:
after changing the wait_for_status() function, I think it will be better to apply the same change here:
not returning anything if succeed and raise timeoutException on failure

Reply:
And add the assertion msg to the function itself

Reply:
done